### ISSUE TITLE
Embed IRCCloud pastebins

### ISSUE THREAD

**Issue by Techman-** [glowing-bear/glowing-bear#972]
I'd really love if Glowing-Bear could embed IRCCloud pastebins. However, as of right now I don't think IRCCloud has any sort of easily-pluggable embed script like Twitter, etc. do, so the frame and everything that goes around the pastebin is rendered directly.

I reached out to James about this and will update when I get a response.

**Comment by Techman-** [comment]
>02:48:35 AM <`@james`> Techman: we fetch the contents via the JSON endpoint and then use ace to render it with syntax highlights in browser.

-------------------------------------
So, embedding their pastebins will be a lot more tedious because they don't have a standard embed API like Pastebin.com has. This is above my experience with JavaScript (I didn't know what Ace was until @jwheare mentioned it), so someone else will have to look at this.

**Comment by lorenzhs** [comment]
Well I mean embedding un-highlighted content would be an improvement over the status quo, adding a syntax highlighting dependency for irccloud pastes sounds a bit iffy to me...

**Comment by Techman-** [comment]
Well, IRCCloud pastebin embeds have a toggle option to enable syntax highlighting, so I think that should be loaded if a user decides to embed the pastebin.

I do not know if @jwheare plans on making an API otherwise, so this is pretty much what the situation is. I wouldn't want half-arsed pastebin embedding because it just looks sloppy.

**Comment by lorenzhs** [comment]
Yeah I mean feel free to implement the full version, if it's done properly and loads the highlighting code only once and on demand (when an irccloud pastebin is first expanded in syntax-highlighting mode) then I think that would be fine with us. Syntax highlighting is nice but *not essential to a pastebin*, that's why I suggested going the easy route. I don't think it's half-arsed, and I don't see why you would prefer no irccloud pastes over non-syntax-highlighted pastes, but if you want to implement the whole thing then feel free to do so.

**Comment by Techman-** [comment]
I don't know JavaScript enough to implement something like this, nor do I have experience working with Angular.

`¯\_(ツ)_/¯`

**Comment by markuman** [comment]
#286
